yamllint/rules/new_lines.py

```diff
@@ -47,7 +52,12 @@ def check(conf, line):
                     line.buffer[line.end:line.end + 2] != '\r\n'):
                 yield LintProblem(1, line.end - line.start + 1,
                                   'wrong new line character: expected \\r\\n')
-        else:
-            if line.buffer[line.end] == '\r':
+        elif conf['type'] == 'unix':
+            if line.buffer[line.end] != '\n':
                 yield LintProblem(1, line.end - line.start + 1,
                                   'wrong new line character: expected \\n')
+        elif conf['type'] == 'auto':
+            if line.buffer[line.end:line.end + len(os_linesep)] != os_linesep:
+                yield LintProblem(1, line.end - line.start + 1,
+                                  'wrong new line character: expected {}'
+                                  .format(repr(os_linesep)))
```

Comment:
I just realised that this line result in `wrong new line character: expected '\r\n'` where as the other option don't have the `'`, should I crop it, eithr with `.format(repr(os.linesep)[1:-1]))` or `.format(repr(os.linesep).strip('\'')))` or just keep it?

Reply:
I would prefer implementing my proposition at https://github.com/adrienverge/yamllint/pull/474#discussion_r901639536, which doesn't involve modifying user messages.